### PR TITLE
CAR 6008 - Temporarily disable failing unit tests in carma-platform's CI

### PR DESCRIPTION
**Pull request by saina-ramyar**
# PR Details
## Description

Temporarily disabled old failing unit tests, to be able to use CI for new development

## Related Jira Key

https://usdot-carma.atlassian.net/browse/CAR-6008

https://github.com/usdot-fhwa-stol/carma-platform/issues/2335

## Motivation and Context

urrently, the CI pipeline in carma-platform does not complete because unit tests are failing. Because of this, Sonar static analysis does not run on new code. 

 

The goal of this PR is to temporarily disable (i.e. comment out) these failing unit tests. During this effort, some time should be spent documenting [1] if a failing unit test could easily be fixed, [2] if the test should be removed completely, [3] other important notes.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp [hunk 1]
```diff
@@ -175,6 +175,10 @@ namespace approaching_emergency_vehicle_plugin{
         ASSERT_EQ(worker_node->filter_points_ahead(reference_point6, {points.back()}).size(), 1);
     }
 
+    // These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+    // Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+
+    /**
```
Comment (JonSmet) [inline approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp:181]: I thought I'd fixed the issues with this package in a recent PR: https://github.com/usdot-fhwa-stol/carma-platform/pull/2322

Just curious, did CI show that this was failing again, or was it commented-out after reviewing the old unit tests logs posted to the Jira story? If these were commented out due to the older unit test log posted on Jira, we might be able to uncomment these now

File: carma_cooperative_perception/CMakeLists.txt [hunk 2]
```diff
@@ -133,8 +133,10 @@ if(carma_cooperative_perception_BUILD_TESTS)
   )
 
   add_launch_test(test/track_list_to_external_object_list_launch_test.py)
-  add_launch_test(test/multiple_object_tracker_duplicates_launch_test.py)
-  add_launch_test(test/multiple_object_tracker_static_obstacle_launch_test.py)
+  # This test has been temporarily disabled to support Continuous Improvement (CI) processes.
+  # Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+  # add_launch_test(test/multiple_object_tracker_duplicates_launch_test.py)
+  # add_launch_test(test/multiple_object_tracker_static_obstacle_launch_test.py)
```
Comment (JonSmet) [inline carma_cooperative_perception/CMakeLists.txt:139]: @MishkaMN @adamlm do we want to prioritize fixing these unit tests as part of the release for VRU?

File: carma_cooperative_perception/test/test_msg_conversion.cpp [hunk 3]
```diff
@@ -54,7 +54,7 @@ TEST(ToTimeMsg, NulloptSeconds)
   EXPECT_DOUBLE_EQ(actual_msg.sec, expected_msg.sec);
   EXPECT_DOUBLE_EQ(actual_msg.nanosec, expected_msg.nanosec);
 }
-
+/**
```
Comment (JonSmet) [inline carma_cooperative_perception/test/test_msg_conversion.cpp:57]: Let's add the `This test has been temporarily disabled ...` here as well

File: mobilitypath_publisher/test/test_mobility_path_publisher.cpp [hunk 4]
```diff
@@ -22,6 +22,7 @@
 
 #include "mobilitypath_publisher/mobilitypath_publisher.hpp"
 
+/**
```
Comment (JonSmet) [inline mobilitypath_publisher/test/test_mobility_path_publisher.cpp:25]: Let's add the `These tests have been temporarily disabled ...` here as well

File: motion_computation/CMakeLists.txt [hunk 5]
```diff
@@ -78,9 +78,10 @@ if(motion_computation_BUILD_TESTS)
     PRIVATE
       -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG
   )
-
-  add_launch_test(test/motion_computation_external_object_launch_test.py)
-  add_launch_test(test/motion_computation_queueing_launch_test.py)
+  # These launch tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+  # Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+  # add_launch_test(test/motion_computation_external_object_launch_test.py)
+  # add_launch_test(test/motion_computation_queueing_launch_test.py)
```
Comment (JonSmet) [inline motion_computation/CMakeLists.txt:84]: @MishkaMN @adamlm do we want to prioritize fixing these unit tests as part of the release for VRU?

File: plan_delegator/test/test_plan_delegator.cpp [hunk 6]
```diff
@@ -25,6 +25,9 @@
 
 namespace plan_delegator{
 
+    // These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+    // Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+    /**
```
Comment (JonSmet) [inline plan_delegator/test/test_plan_delegator.cpp:30]: Is every unit test failing for plan_delegator currently?

File: platooning_control_ihp/CMakeLists.txt [hunk 7]
```diff
@@ -118,10 +118,11 @@ if(TARGET ${PROJECT_NAME}-test)
...
-if(CATKIN_ENABLE_TESTING)
-  find_package(rostest REQUIRED)
-  add_rostest_gtest(test_platoon_control_ihp test/mynode.test test/test_mynode.cpp)
-  target_link_libraries(test_platoon_control_ihp ${catkin_LIBRARIES})
-endif()
-
-
+# These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+# Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+#TODO Temporary because commenting out the tests didnt work
```
Comment (JonSmet) [inline platooning_control_ihp/CMakeLists.txt:123]: Just placing a comment so we remember to resolve this TODO comment either in this PR or track it in a separate GitHub issue.

File: route/test/test_route_generator.cpp [hunk 8]
```diff
@@ -35,6 +35,10 @@
 #include "route/route_state_worker.hpp"
 #include "route/route_node.hpp"
 
+// These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+// Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+
+/**
```
Comment (JonSmet) [inline route/test/test_route_generator.cpp:41]: I thought I'd fixed the issues with this package in a recent PR: https://github.com/usdot-fhwa-stol/carma-platform/pull/2301

Just curious, did CI show that this was failing again, or was it commented-out after reviewing the old unit tests logs posted to the Jira story? If these were commented out due to the older unit test log posted on Jira, we might be able to uncomment these now.

File: yield_plugin/test/test_yield_plugin.cpp [hunk 9]
```diff
@@ -43,12 +43,16 @@ class YieldNode :  public carma_ros2_utils::CarmaLifecycleNode
...
       RCLCPP_INFO(get_logger(),"Test lc callback..");
     }
-    
+
 };
 
+// These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+// Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+
+/**
```
Comment (JonSmet) [inline yield_plugin/test/test_yield_plugin.cpp:52]: @MishkaMN @adamlm do we want to prioritize fixing these unit tests as part of the release for VRU?

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: .github/workflows/Ci.yml [hunk 10]
```diff
@@ -58,9 +58,14 @@ jobs:
     - name: Run ROS1 C++ Tests
       # Run ROS1 C++ tests
       # The skipped ROS 1 packages here should match the ROS 1 packages that are skipped during the build in docker/install.sh"
+
+      # These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+      # Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
+      # TODO carma_utils ros test doesnt run sometimes, so temporatily disabled
```
[the remark this answers is not in the document]
Comment (saina-ramyar) [reply .github/workflows/Ci.yml:64]: Based on the CI logs it seemed like the ROS tests don't even start, they fail in the beginning like a timeout.
I tried to replicate it locally, but on my PC the tests always pass.
Even on the CI it sometimes run successfully, but sometimes it fails.
I think we need further investigation to find the cause.

File: approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp (hunk 1 above)
Comment (saina-ramyar) [reply approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp:181]: Did another test, testBSMProcessing still fails, but the other one passed so I enabled it again

File: carma_cooperative_perception/test/test_msg_conversion.cpp (hunk 3 above)
Comment (saina-ramyar) [reply carma_cooperative_perception/test/test_msg_conversion.cpp:57]: added

File: mobilitypath_publisher/test/test_mobility_path_publisher.cpp (hunk 4 above)
Comment (saina-ramyar) [reply mobilitypath_publisher/test/test_mobility_path_publisher.cpp:25]: added

File: plan_delegator/test/test_plan_delegator.cpp (hunk 6 above)
Comment (saina-ramyar) [reply plan_delegator/test/test_plan_delegator.cpp:30]: It seemed that way initially, but I ran locally and only the last few failed. Uncommented the rest.

File: route/test/test_route_generator.cpp (hunk 8 above)
Comment (saina-ramyar) [reply route/test/test_route_generator.cpp:41]: yes, this was the first package I updated and just the list. Uncommented the tests.

File: .github/workflows/Ci.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (JonSmet) [reply .github/workflows/Ci.yml:64]: Okay that makes sense, thanks for the explanation. If you agree, can we remove the `TODO` comment here, and then I think we could include in the specific notes on the `carma_utils` test failures either in the GitHub Issue directly, or in our separate document that will eventually be created that includes notes on all failing unit tests (part of CAR-6009)?

File: approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp [hunk 11]
```diff
@@ -856,19 +861,21 @@ namespace approaching_emergency_vehicle_plugin{
...
-        ASSERT_EQ(resp3->new_plan.maneuvers[0].lane_following_maneuver.lane_ids, maneuver_0_lane_id); 
+        ASSERT_EQ(resp3->new_plan.maneuvers[0].lane_following_maneuver.lane_ids, maneuver_0_lane_id);
         ASSERT_EQ(resp3->new_plan.maneuvers[0].lane_following_maneuver.parameters.negotiation_type, carma_planning_msgs::msg::ManeuverParameters::NO_NEGOTIATION);
         ASSERT_EQ(resp3->new_plan.maneuvers[0].lane_following_maneuver.parameters.planning_strategic_plugin, "approaching_emergency_vehicle_plugin");
         ASSERT_EQ(resp3->new_plan.maneuvers[0].lane_following_maneuver.parameters.presence_vector, carma_planning_msgs::msg::ManeuverParameters::HAS_TACTICAL_PLUGIN);
         ASSERT_EQ(resp3->new_plan.maneuvers[0].lane_following_maneuver.parameters.planning_tactical_plugin, "inlanecruising_plugin");
     }
 
+    // These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+    // Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
```
Comment (JonSmet) [inline approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp:878]: Looks like we can remove this comment now that `testWarningBroadcast` is no longer commented out.

File: plan_delegator/test/test_plan_delegator.cpp [hunk 12]
```diff
@@ -25,6 +25,10 @@
 
 namespace plan_delegator{
 
+    // These tests has been temporarily disabled to support Continuous Improvement (CI) processes.
+    // Related GitHub Issue: <https://github.com/usdot-fhwa-stol/carma-platform/issues/2335>
```
Comment (JonSmet) [inline plan_delegator/test/test_plan_delegator.cpp:29]: Looks like we need to move this comment down now to the two tests that are commented out

File: stop_and_dwell_strategic_plugin/test/test_stop_and_dwell_strategic_plugin.cpp [hunk 13]
```diff
@@ -36,12 +36,12 @@ class DummyNode :  public carma_ros2_utils::CarmaLifecycleNode
     carma_ros2_utils::CallbackReturn handle_on_configure(const rclcpp_lifecycle::State &){return CallbackReturn::SUCCESS;}
 };
 
-
+/**
```
Comment (JonSmet) [inline stop_and_dwell_strategic_plugin/test/test_stop_and_dwell_strategic_plugin.cpp:39]: Can we add the comment here for this commented out test?

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: .github/workflows/Ci.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (saina-ramyar) [reply .github/workflows/Ci.yml:64]: I added the details on the document with the list of disabled unit tests as well. But I think its good to add it to the issue as well. I just noticed same thing happened with stop_and_dwell, and maybe it was the case with some other packages as well, meaning they randomly fail later.

File: approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp (hunk 11 above)
Comment (saina-ramyar) [reply approaching_emergency_vehicle_plugin/test/test_approaching_erv.cpp:878]: removed

File: plan_delegator/test/test_plan_delegator.cpp (hunk 12 above)
Comment (saina-ramyar) [reply plan_delegator/test/test_plan_delegator.cpp:29]: removed

File: stop_and_dwell_strategic_plugin/test/test_stop_and_dwell_strategic_plugin.cpp (hunk 13 above)
Comment (saina-ramyar) [reply stop_and_dwell_strategic_plugin/test/test_stop_and_dwell_strategic_plugin.cpp:39]: added

File: platooning_control_ihp/CMakeLists.txt (hunk 7 above)
Comment (saina-ramyar) [reply platooning_control_ihp/CMakeLists.txt:123]: added a note in the document

**Review by JonSmet — APPROVED** [review]

**Pull request merged**